File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-0.java
class Solution {
    private int maxSum = Integer.MIN_VALUE;

    public int maxPathSum(TreeNode root) {
        dfs(root);
        return maxSum;
    }

    private int dfs(TreeNode node) {
        if (node == null) return 0;

        int left = Math.max(0, dfs(node.left));
        int right = Math.max(0, dfs(node.right));

        // Path passing through this node
        int currentPath = node.val + left + right;
        maxSum = Math.max(maxSum, currentPath);

        // Return max path going down
        return node.val + Math.max(left, right);
    }
}

File: Data Structures & Algorithms/longest-turbulent-subarray/submission-0.cpp
class Solution {
public:
    int maxTurbulenceSize(vector<int>& arr) {
        int n = arr.size();
        if (n == 1) return 1;

        int ans = 1, left = 0;

        for (int right = 1; right < n; right++) {
            int c = (arr[right - 1] > arr[right]) ? 1 :
                    (arr[right - 1] < arr[right]) ? -1 : 0;

            if (c == 0) {
                left = right;
            } else if (right == n - 1 || 
                      c * ((arr[right] > arr[right + 1]) ? 1 :
                           (arr[right] < arr[right + 1]) ? -1 : 0) != -1) {
                ans = max(ans, right - left + 1);
                left = right;
            }
        }

        return ans;
    }
};